db/connect.js

```diff
@@ -0,0 +1,42 @@
+import mongoose from "mongoose";
+
+const MONGODB_URI = process.env.MONGODB_URI;
+
+if (!MONGODB_URI) {
+  throw new Error(
+    "Please define the MONGODB_URI environment variable inside .env.local"
+  );
+}
+
+let cached = global.mongoose;
+
+if (!cached) {
+  cached = global.mongoose = { conn: null, promise: null };
+}
+
+async function dbConnect() {
+  if (cached.conn) {
+    return cached.conn;
+  }
+
+  if (!cached.promise) {
+    const opts = {
```

Comment:
Maybe options is better than using abbreviation.
```suggestion
    const options = {
```

Reply:
Haha, you're right! That's the `dbconnect`, I actually just copied and pasted it from other projects and didn't edit it any further. But I've now written `options` (and also `error` instead of `e`). Thanks for the tip.